ex2_system/source/diagnostic/diagnosic.c

```diff
@@ -54,22 +54,24 @@ static void uhf_watchdog_daemon(void *pvParameters) {
         if (err != U_GOOD_CONFIG) {
             ex2_log("UHF was not responsive - attempting to toggle power.");
             // Toggle the UHF.
-            const unsigned int timeout = 30 * ONE_SECOND; // 30 seconds
+            const unsigned int timeout = 5 * ONE_SECOND; // 5 seconds
+            unsigned int attempts = 0;
             eps_set_pwr_chnl(UHF_PWR_CHNL, OFF);
             TickType_t start = xTaskGetTickCount();
 
-            while (eps_get_pwr_chnl(UHF_PWR_CHNL) != OFF && xTaskGetTickCount() - start < timeout) {
+            while ((eps_get_pwr_chnl(UHF_PWR_CHNL) != OFF) || ((xTaskGetTickCount() - start) < timeout)) {

```

Comment:
No. If the channel is OFF, we want to break. This says that we should keep going until the timeout expires regardless of the channel, but really we only want to retry as long as the channel is in the wrong state.

Reply:
Not really. We want the system to remain off until the timeout expires. This is because often the capacitances on systems are high enough that it takes a number of seconds for them to fully deenergize. That is, if we turn it off, and immediately turn it back on, some of the caps will be able to keep some components functional and a full power reset might not occur. By forcing the system to turn off for at least 5 seconds we avoid this issue. Thomas and I agree that this is a better way to watchdog.

Reply:
I agree that it could be written better, though. Particularly considering your next commend. I'll take another look at it.